### DISCUSSION TITLE
Prevent creating a certain a path in getStaticPaths

### DISCUSSION THREAD

**Discussion by hasanm95** [vercel/next.js#38253] [Ideas]
I created dynamic pages by getStaticPaths and getStaticProps. 

/blogs/page/[page]  
├   ├ /blogs/page/1
├   ├ /blogs/page/2
├   ├ /blogs/page/3
├   └ /blogs/page/4

But here I don't want to create the page/1. Is there any way to prevent this page from creating?

**Comment by icyJoseph** [comment]
You should filter it out from the `getStaticPaths` result.

And also, inside `getStaticProps`, if you are using ISR, you should check that if `page` is one, `{ notFound: true }` should be returned. 

Or if it is the case that you don't want to pre-render it, but what to render it after deploy then, you don't need to do the last bit.

```tsx
export async function getStaticPaths(): Promise<GetStaticPathsResult> {
  const pages: number[] = await getAllPages(); // or whatever you do here to get this value
  // early on filter it out
  const pagesToGenerate = pages.filter((page) => page !== 1);
  // these filter + map pass could be chained, or replaced by a loop
  const paths = pagesToGenerate.map((page) => {
    return { params: { page: `${page}` } }; // cast page to string
  });

  return {
    paths,
    fallback: true, // false or 'blocking'
  };
}
```

I suspect, perhaps you already tried this? is there anything else I may be missing?

**Reply by hasanm95** [reply]
Thanks. It worked

**Reply by icyJoseph** [reply]
Cool! Don't forget to mark as answered!
